cartography/intel/azure/sql.py

```diff
@@ -277,267 +291,237 @@ def load_server_details(neo4j_session, credentials, subscription_id, details, up
+) -> None:
     """
     Ingest the restorable dropped database details into neo4j.
     """
     ingest_restorable_dropped_databases = """
-    MERGE (rdd:AzureRestorableDroppedDatabase{id: {DatabaseId}})
+    UNWIND {restorable_dropped_databases_list} as res_dropped_dbs
+    MERGE (rdd:AzureRestorableDroppedDatabase{id: res_dropped_dbs.id})
     ON CREATE SET rdd.firstseen = timestamp(), rdd.lastupdated = {azure_update_tag}
-    SET rdd.name = {Name},
-    rdd.location = {Location},
-    rdd.databasename = {DatabaseName},
-    rdd.creationdate = {CreationDate},
-    rdd.deletiondate = {DeletionDate},
-    rdd.restoredate = {RestoreDate},
-    rdd.edition = {Edition},
-    rdd.servicelevelobjective = {ServiceLevelObjective},
-    rdd.maxsizebytes = {MaxSizeBytes}
-    WITH rdd
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET rdd.name = res_dropped_dbs.name,
+    rdd.location = res_dropped_dbs.location,
+    rdd.databasename = res_dropped_dbs.database_name,
+    rdd.creationdate = res_dropped_dbs.creation_date,
+    rdd.deletiondate = res_dropped_dbs.deletion_date,
+    rdd.restoredate = res_dropped_dbs.earliest_restore_date,
+    rdd.edition = res_dropped_dbs.edition,
+    rdd.servicelevelobjective = res_dropped_dbs.service_level_objective,
+    rdd.maxsizebytes = res_dropped_dbs.max_size_bytes
+    WITH rdd, res_dropped_dbs
+    MATCH (s:AzureSQLServer{id: res_dropped_dbs.server_id})
     MERGE (s)-[r:RESOURCE]->(rdd)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for data in restorable_dropped_databases:
-        neo4j_session.run(
-            ingest_restorable_dropped_databases,
-            DatabaseId=data['id'],
-            Name=data['name'],
-            Location=data['location'],
-            DatabaseName=data['database_name'],
-            CreationDate=data['creation_date'],
-            DeletionDate=get_optional_value(data, ['deletion_date']),
-            RestoreDate=data['earliest_restore_date'],
-            Edition=data['edition'],
-            ServiceLevelObjective=data['service_level_objective'],
-            MaxSizeBytes=data['max_size_bytes'],
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_restorable_dropped_databases,
+        restorable_dropped_databases_list=restorable_dropped_databases,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_failover_groups(neo4j_session, failover_groups, update_tag):
+def _load_failover_groups(
+        neo4j_session: neo4j.Session, failover_groups: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the failover groups details into neo4j.
     """
     ingest_failover_groups = """
-    MERGE (f:AzureFailoverGroup{id: {GroupId}})
+    UNWIND {failover_groups_list} as fg
+    MERGE (f:AzureFailoverGroup{id: fg.id})
     ON CREATE SET f.firstseen = timestamp(), f.lastupdated = {azure_update_tag}
```

Comment:
lastupdated should not be in the create set. it should be in update set.

Reply:
done